server/commands/connected_objects.go

```diff
@@ -21,10 +21,58 @@ type ConnectedObjectsRequest struct {
 	StoreID          string
 	ObjectType       string
 	Relation         string
-	User             *openfgapb.ObjectRelation
+	User             isUserRef
 	ContextualTuples []*openfgapb.TupleKey
 }
 
+type isUserRef interface {
```

Comment:
since the conncted_objects.go file is getting large and the isUserRef/UserRefObject etc. are used by both connected_objects.go and list_objects.go, will it make sense to move them to a separate file and keep this file to be strictly regarding ConnectedObjectsCommand?

Reply:
Realistically this should belong in our protobuf files. Such as

```
message User {
    oneof user_ref {
        Object object = 1;
        TypedWildcard typed_wildcard = 2;
        ObjectRelation object_relation = 3;
    }
}

message Object {
    string type = 1;
    string id = 2;
}

message TypedWildcard {
    string type = 1;
}

message ObjectRelation {
    Object object = 1;
    string relation = 2;
}
```

Reply:
Because of the scope of impact of these changes, I'd like to do this promptly in a follow up PR. Changing it in the scope of this PR has some pretty broad impact.